pkg/ruler/manager.go

```diff
@@ -54,10 +54,7 @@ func NewDefaultMultiTenantManager(cfg Config, managerFactory ManagerFactory, reg
 		return nil, err
 	}
 
-	userManagerMetrics := NewManagerMetrics()
-	if reg != nil {
-		reg.MustRegister(userManagerMetrics)
-	}
+	userManagerMetrics := NewManagerMetrics(reg)
```

Comment:
You can now initialise it directly below:

```
userManagerMetrics: NewManagerMetrics(reg),
``` 

Reply:
Do we want this New(reg) pattern or not? @pstibrany says no

Reply:
Sorry @roidelapluie for the long reply. Me and Peter had a discussion offline. We have a different opinion here. I commit to Peter's take: given `ManagerMetrics` is a `Collector`, and a `Collector` shouldn't take the registerer, let's not pass the registerer to `NewManagerMetrics()`.

_Same applies to alertmanager metrics._